profiles/car.lua

```diff
@@ -606,9 +701,45 @@ function way_function (way, result)
       result.backward_speed = result.backward_speed * scaling_factor
     end
   end
+end
 
+-- determine if this way can be used as a start/end point for routing
+function handle_startpoint(way,result)
   -- only allow this road as start point if it not a ferry
-  result.is_startpoint = result.forward_mode == mode.driving or result.backward_mode == mode.driving
+  result.is_startpoint = result.forward_mode == mode.driving or 
+                              result.backward_mode == mode.driving
+end
+
+-- set the road classification based on guidance globals configuration
+function handle_classification(way,result)
+  set_classification(way:get_value_by_key("highway"),result,way)
+end
+
+-- main entry point for processsing a way
+function way_function(way, result)
+  -- table of temporary values, e.g. computed access status
+  local data = {}
+  
+  -- perform each procesing step sequentially.
+  -- most steps can abort processing, meaning the way
+  -- is not routable  
+  
+  if handle_initial_check(way,result,data) == false then return end
```

Comment:
this function could probably be not any more ambiguous. What is a initial check? What are we filtering here?